autosklearn/pipeline/components/data_preprocessing/text_encoding/bag_of_word_encoding_distinct.py

```diff
@@ -0,0 +1,118 @@
+from typing import Dict, Optional, Tuple, Union
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+import ConfigSpace.hyperparameters as CSH
+from ConfigSpace import EqualsCondition
+
+import numpy as np
+import pandas as pd
+from scipy.sparse import hstack
+
+from autosklearn.pipeline.base import DATASET_PROPERTIES_TYPE, PIPELINE_DATA_DTYPE
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
+
+from sklearn.feature_extraction.text import CountVectorizer
+
+
+class BagOfWordEncoder(AutoSklearnPreprocessingAlgorithm):
+    def __init__(
+        self,
+        ngram_range: int = 1,
+        min_df_choice: str = "min_df_absolute",
+        min_df_absolute: int = 0,
+        min_df_relative: float = 0.01,
+        random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        self.ngram_range = ngram_range
+        self.random_state = random_state
+        self.min_df_choice = min_df_choice
+        self.min_df_absolute = min_df_absolute
+        self.min_df_relative = min_df_relative
+
+    def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
+            ) -> 'BagOfWordEncoder':
+
+        if isinstance(X, pd.DataFrame):
+            if self.min_df_choice == "min_df_absolute":
+
+                self.preprocessor = {}
+
+                for feature in X.columns:
+                    vectorizer = CountVectorizer(min_df=self.min_df_absolute,
+                                                 ngram_range=(1, self.ngram_range)).fit(X[feature])
+                    self.preprocessor[feature] = vectorizer
+
+            elif self.min_df_choice == "min_df_relative":
+
+                self.preprocessor = {}
+
+                for feature in X.columns:
+                    vectorizer = CountVectorizer(min_df=self.min_df_relative,
+                                                 ngram_range=(1, self.ngram_range)).fit(X[feature])
+                    self.preprocessor[feature] = vectorizer
+            else:
+                raise KeyError()
+        else:
+            raise ValueError("Your text data is not encoded in a pandas.DataFrame\n"
+                             "Please make sure to use a pandas.DataFrame and ensure"
+                             "that the text features are encoded as strings.")
+        return self
+
+    def transform(self, X: PIPELINE_DATA_DTYPE) -> PIPELINE_DATA_DTYPE:
+        X_new = None
+        if self.preprocessor is None:
+            raise NotImplementedError()
+
+        for feature in self.preprocessor:
+            # the names in the dataframe must not change
+            if X_new is None:
+                X_new = self.preprocessor[feature].transform(X[feature])
```

Comment:
I guess `dropna` is needed here too, in general where ever we query a column from `X`